### PR TITLE
Staging Sites: Customize Privacy section to streamline 'Public' option

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes https://github.com/Automattic/dotcom-forge/issues/2105
Related https://github.com/Automattic/wpcomsh/pull/1343

## Proposed Changes

Streamlines the 'Public' option for the Privacy Settings of a Staging Site:

image

Non-Staging Sites remain the same:

image

Jetpack Sites remain the same:

image

## Testing Instructions

1. Create a Business site and take it Atomic.
2. Navigate to 'General Settings' and verify the 'Launch site' UI appears as expected.
3. Launch the site and verify the Privacy settings appear as expected (unchanged).
4. Navigate to 'Hosting Configuration' and create a staging site.
5. Navigate to 'General Settings' for the staging site and verify the new Privacy Settings appear as expected.
6. Spin up a jurassic.ninja site and verify the Privacy Settings appear as expected (unchanged).
7. Create a Personal site and verify the Privacy Settings appear as expected (unchanged).

### CODE REVIEW

File: client/my-sites/site-settings/form-general.jsx [hunk 1]
```diff
@@ -414,7 +415,7 @@ export class SiteSettingsFormGeneral extends Component {
 							) }
 						</>
 					) }
-				{ ! isNonAtomicJetpackSite && (
+				{ ! isNonAtomicJetpackSite && ! isWpcomStagingSite && (
```
Comment (sejas) [inline client/my-sites/site-settings/form-general.jsx]: Could we re-use the same `FormLabel`/`FormRadio` independently if `isWpcomStagingSite`?

Something like this:
```suggestion
				{ ! isNonAtomicJetpackSite && (
```

```diff
					<FormLabel className="site-settings__visibility-label is-public">
						<FormRadio
							name="blog_public"
							value="1"
							checked={
								( wpcomPublicComingSoon && blogPublic === 0 && isComingSoonDisabled ) ||
								( blogPublic === 0 && ! wpcomPublicComingSoon ) ||
								blogPublic === 1
							}
							onChange={ () =>
								this.handleVisibilityOptionChange( {
-									blog_public: 1,
+									blog_public: isWpcomStagingSite ? 0 : 1,
									wpcom_coming_soon: 0,
									wpcom_public_coming_soon: 0,
								} )
							}
							disabled={ isRequestingSettings }
							onClick={ eventTracker( 'Clicked Site Visibility Radio Button' ) }
							label={ translate( 'Public' ) }
						/>
					</FormLabel>
```

File: client/my-sites/site-settings/form-general.jsx [hunk 2]
```diff
@@ -437,34 +438,69 @@ export class SiteSettingsFormGeneral extends Component {
...
+								label={ translate( 'Public' ) }
+							/>
+						</FormLabel>
+						<FormSettingExplanation>
+							{ translate(
+								'Your site is visible to everyone, but search engines are discouraged from indexing. Staging sites cannot be indexed by search engines.'
+							) }
+						</FormSettingExplanation>
+					</>
+				) }
```
Comment (sejas) [inline client/my-sites/site-settings/form-general.jsx]: Then we could avoid duplicating all this logic:

```suggestion
				{ isWpcomStagingSite && (
					<FormSettingExplanation>
						{ translate(
							'Your site is visible to everyone, but search engines are discouraged from indexing. Staging sites cannot be indexed by search engines.'
						) }
					</FormSettingExplanation>
				) }
```

**Review by sejas — CHANGES_REQUESTED** [review]
@danielbachhuber , I think we could optimize the code a bit.
What do you think?

It works as expected. The actual value of `blog_public` is overwrited in the backend, but I agree that it's more consistent to change it in the frontend too.

We could consider adding new tests.

**Comment by javierarce** [comment]
I like the general approach, but here's an alternative copy for the new option:

`Your staging site is visible to everyone but search engines won't index it.`

If we say "your staging site" here we should probably do the same in the other options… so another possibility could be:

`Your site is visible to everyone, but as a staging site search engines won't index it.`

**Comment by danielbachhuber** [comment]
I went with `Your site is visible to everyone, but search engines are discouraged from indexing staging sites.`.

The form seems a bit boogered right now, though 😞 I think I need to give this a really thorough shake with lots of tests.

**Review by kriskorn — APPROVED** [review]
Looks good. Thank you!

**Review by sejas — APPROVED** [review]
@danielbachhuber, Thanks for adding the `<PublicFormRadio />` component.

It works as described.

Screenshot 2023-04-25 at 13 55 25

**Comment by danielbachhuber** [comment]
@kriskorn @vparkhere Could one of you give this some thorough testing in the calypso.live environment? More specifically, testing with Business + staging sites, but also with Jetpack-connected sites, Simple sites, and maybe P2s. You can compare the behavior in calypso.live with wordpress.com.

**Comment by kriskorn** [comment]
@danielbachhuber We will do that in a couple of hours and report back afterwards, so expect a comment from me or Victoria in about 4-6 hours.

**Review by vparkhere — APPROVED** [review]
Checked:
1) Jurassic Ninja
2) Atomic
3) Business simple plan 
4) free simple site
5) staging site

Everything looked and worked as expected!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7949009

Thank you @danielbachhuber for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7949009).

**Pull request merged**